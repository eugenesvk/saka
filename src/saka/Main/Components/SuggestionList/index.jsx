import 'material-components-web/dist/material-components-web.css';
import 'scss/styles.scss';
import { h, Component } from 'preact';
import Suggestion from './Containers/SuggestionSelector.jsx';

export default ({
  searchString,
  suggestions,
  selectedIndex,
  firstVisibleIndex,
  maxSuggestions,
  onSuggestionClick
}) => (
  <ul className="mdc-deprecated-list mdc-deprecated-list--two-line mdc-deprecated-list--avatar-list two-line-avatar-text-icon-demo list-container">
    {suggestions
      .slice(firstVisibleIndex, firstVisibleIndex + maxSuggestions)
      .map((suggestion, index) => {
        return (
          <Suggestion
            suggestion={suggestion}
            searchString={searchString}
            selected={index === selectedIndex}
            index={index}
            onClick={onSuggestionClick}
          />
        );
      })}
  </ul>
);
