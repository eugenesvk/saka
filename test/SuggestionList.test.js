import SuggestionList from '@/saka/Main/Components/SuggestionList/index.jsx';
import { render } from '@testing-library/preact';
import { h } from 'preact';

const MAX_SUGGESTIONS = 6;

beforeEach(() => {
  // Clears the database and adds some testing data.
  // Jest will wait for this promise to resolve before running tests.
  global.SAKA_PLATFORM = 'chrome';
});

describe('SuggestionList component ', () => {
  it('should be empty when no values provided', () => {
    const suggestions = [];

    const searchString = {};

    const { container } = render(
      <SuggestionList
        searchString={searchString}
        suggestions={suggestions}
        selectedIndex={0}
        firstVisibleIndex={0}
        maxSuggestions={MAX_SUGGESTIONS}
      />
    );

    expect(container).toMatchSnapshot();
  });

  it('should display suggestions when provided', () => {
    const suggestions = [
      {
        type: 'tab',
        title: 'lusakasa/saka: Elegant tab search',
        url: 'https://github.com/lusakasa/saka'
      },
      {
        type: 'tab',
        title: 'Google',
        url: 'https://google.com'
      }
    ];

    const searchString = '';

    const { getByText } = render(
      <SuggestionList
        searchString={searchString}
        suggestions={suggestions}
        selectedIndex={0}
        firstVisibleIndex={0}
        maxSuggestions={MAX_SUGGESTIONS}
        onSuggestionClick={() => {}}
      />
    );

    suggestions.map(suggestion => {
      getByText(suggestion.title);
      getByText(suggestion.url);
    });
  });
});
