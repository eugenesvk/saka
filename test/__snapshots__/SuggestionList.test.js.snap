// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuggestionList component  should be empty when no values provided 1`] = `
<div>
  <ul
    class="mdc-deprecated-list mdc-deprecated-list--two-line mdc-deprecated-list--avatar-list two-line-avatar-text-icon-demo list-container"
  />
</div>
`;
